binary-search-tree/node-js/inventory-management.js

```diff
@@ -18,16 +18,37 @@ class BinarySearchTree {
         this.root = null;
     }
 
-    search(id) {
-        console.error("you must implement search");
+    search(id, currentNode) {
+        if (id === null) {
```

Comment:
I get what you're trying to do here by making sure we only process valid input, but I think you can reasonable always assume that `id` will be a valid value whether or not a node with the same value exists in the binary search tree.